labs/src/main/scala/org/scalalabs/basic/lab01/OOExercise.scala

```diff
@@ -40,6 +40,29 @@ import scala.language.implicitConversions
  *   of type [[org.scalalabs.basic.lab01.CurrencyConverter]]
  * - Use the implicit CurrencyConverter to do the conversion. 
  */
-class Euro {
+//Modified by Sisi Cheng & Jean Sung
+class Euro (var euro:Int, val cents:Int = 0) extends Currency {
+	val inCents:Int = euro*100 + cents;
+	symbol = "EUR"
+	
+	def +(rhs:Euro) = Euro.fromCents(this.inCents + rhs.inCents)
+	  
+	def *(factor:Int) = Euro.fromCents(this.inCents * factor)
+	
+	override def toString(): String = {
+	  val centsRep:String = if (cents == 0) "--" else f"$cents%02d"
+	  symbol + ": " + f"$euro%01d" + "," + centsRep;
+	}
+}
 
-}
\ No newline at end of file
+object Euro {
+	def fromCents(cents:Int):Euro = {
+		new Euro(cents/100, cents%100) {
```

Comment:
As the comment to `HelloWorldClassAndObject` said, this is not the standard way of doing things. I *think* (based on Java experience and that comment) that this creates a new object of a new type subclassing `Euro`, instead of a new `Euro` directly. If you just got rid of the braces for this, making it `def fromCents(cents:Int):Euro = new Euro(cents/100, cents%100)`, I *think* that would create a new object with type `Euro`, and *probably* have less overhead.

I’m really not sure, given that I only started learning Scala about a week ago like most other people here. Can anybody else comment on these thoughts?